tests/commands/os/configure.spec.ts

```diff
@@ -30,27 +32,175 @@ import { BalenaAPIMock } from '../../nock/balena-api-mock';
 if (process.platform !== 'win32') {
 	describe('balena os configure', function () {
 		let api: BalenaAPIMock;
-		let tmpPath: string;
+		let tmpDummyPath: string;
+		let tmpMatchingDtJsonPartitionPath: string;
+		let tmpNonMatchingDtJsonPartitionPath: string;
 
 		beforeEach(async () => {
 			api = new BalenaAPIMock();
 			api.expectGetWhoAmI({ optional: true, persist: true });
-			tmpPath = (await tmpNameAsync()) as string;
-			await fs.copyFile('./tests/test-data/dummy.img', tmpPath);
+			tmpDummyPath = (await tmpNameAsync()) as string;
+			await fs.copyFile('./tests/test-data/dummy.img', tmpDummyPath);
+			tmpMatchingDtJsonPartitionPath = (await tmpNameAsync()) as string;
+			await fs.copyFile(
+				'./tests/test-data/mock-jetson-nano-6.0.23.with-boot-partition-12.img',
+				tmpMatchingDtJsonPartitionPath,
+			);
+
+			tmpNonMatchingDtJsonPartitionPath = (await tmpNameAsync()) as string;
+			// Create an image with a device-type.json that mentions a non matching boot partition.
+			// We copy the pre-existing image and modify it, since including a separate one
+			// would add 18MB more to the repository.
+			await fs.copyFile(
+				'./tests/test-data/mock-jetson-nano-6.0.23.with-boot-partition-12.img',
+				tmpNonMatchingDtJsonPartitionPath,
+			);
+			await imagefs.interact(
+				tmpNonMatchingDtJsonPartitionPath,
+				12,
+				async (_fs) => {
+					const readFileAsync = promisify(_fs.readFile);
+					const writeFileAsync = promisify(_fs.writeFile);
+
+					const dtJson = JSON.parse(
+						await readFileAsync('/device-type.json', { encoding: 'utf8' }),
+					);
+					expect(dtJson).to.have.nested.property(
+						'configuration.config.partition',
+						12,
+					);
+					dtJson.configuration.config.partition = 999;
+					await writeFileAsync('/device-type.json', JSON.stringify(dtJson));
+
+					await writeFileAsync(
+						'/os-release',
+						stripIndent`
+							ID="balena-os"
+							NAME="balenaOS"
+							VERSION="6.1.25"
+							VERSION_ID="6.1.25"
+							PRETTY_NAME="balenaOS 6.1.25"
+							DISTRO_CODENAME="kirkstone"
+							MACHINE="jetson-nano"
+							META_BALENA_VERSION="6.1.25"`,
+					);
+				},
+			);
 		});
 
 		afterEach(async () => {
 			api.done();
-			await fs.unlink(tmpPath);
+			await fs.unlink(tmpDummyPath);
+			await fs.unlink(tmpMatchingDtJsonPartitionPath);
+			await fs.unlink(tmpNonMatchingDtJsonPartitionPath);
+		});
+
+		it('should inject a valid config.json file to an image with partition 12 as boot & matching device-type.json ', async () => {
+			api.expectGetApplication();
+			api.expectGetDeviceTypes();
+			// It should not reach to /config or /device-types/v1 but instead find
+			// everything required from the device-type.json in the image.
+			// api.expectGetConfigDeviceTypes();
+			api.expectDownloadConfig();
+
+			const command: string[] = [
+				`os configure ${tmpMatchingDtJsonPartitionPath}`,
+				'--device-type jetson-nano',
+				'--version 6.0.13',
```

Comment:
You can remove this line now that `getOsVersionFromImage()` reads from the image. It also would provide another implicit test.